File: include/event/IEvent.r
#ifndef _IEVENT_R_
#define _IEVENT_R_

#include "IClass.r"

#define CONT_IEVENT \
	CONT_ICLASS \
	void (*zoom) ( void *self, void *view );

struct IEvent {
	CONT_IEVENT
};

void event_zoomOnView ( void *_self, void *view );

#endif

File: src/event/IEvent.c
#include <assert.h>

#include "event/IEvent.r"
#include "event/IEvent.h"

void event_zoomOnView ( void *_self, void *view ) {
	struct IEvent **self = _self;

	assert (*self);
	if ( (*self)->zoom )
		(*self)->zoom ( _self, view );
}
